src/pay/batch_payer.py

```diff
@@ -393,37 +393,39 @@ def attempt_single_batch(self, payment_records, op_counter, dry_run=None):
                 if simulation_status == PaymentStatus.FAIL:
-                    logger.info("Payment to {} script could not be processed (Possible reason: liquidated contract). Skipping. (think about redirecting the payout to the owner address using the maps rules. Please refer to the TRD documentation or to one of the TRD maintainers)"
+                    logger.info("Payment to {} script could not be processed. Possible reason: liquidated contract. Skipping. Think about redirecting the payout to the owner address using the maps rules. Please refer to the TRD documentation or to one of the TRD maintainers."
                                 .format(payment_item.paymentaddress))
                     payment_item.paid = PaymentStatus.AVOIDED
                     continue
                 gas_limit, tx_fee, storage_limit = simulation_results
                 burn_fee = COST_PER_BYTE * storage_limit
                 total_fee = tx_fee + burn_fee
-                # Bound the total (baker and burn) fee by the reward amount in case of KT1 accounts
+
                 if total_fee > FEE_LIMIT_CONTRACTS:
-                    logger.info("Payment to {:s} script requires higher fees than reward amount. Skipping. (Needed fee: {:10.6f} XTZ, Max fee: {:10.6f} XTZ) Either configure a higher fee or redirect to the owner address using the maps rules. Refer to the TRD documentation."
+                    logger.info("Payment to {:s} script requires higher fees than reward amount. Skipping. Needed fee: {:10.6f} XTZ, max fee: {:10.6f} XTZ. Either configure a higher fee or redirect to the owner address using the maps rules. Refer to the TRD documentation."
                                 .format(payment_item.paymentaddress, total_fee / MUTEZ, FEE_LIMIT_CONTRACTS / MUTEZ))
                     payment_item.paid = PaymentStatus.AVOIDED
                     continue
 
-                # Fees within safety limits; Subtract from payment amount
+                # Subtract burn fee from the payment amount
                 orig_pymnt_amnt = pymnt_amnt
-                pymnt_amnt = max(pymnt_amnt - total_fee, 0)  # ensure not less than 0
-                logger.info("Payment to {} script requires {:.0f} gas * {:.2f} mutez-per-gas + {:10.6f} default fee = {:10.6f} total fees; Payment reduced from {:10.6f} to {:10.6f}".format(
-                            payment_item.paymentaddress, gas_limit, MUTEZ_PER_GAS_UNIT, self.default_fee / MUTEZ, total_fee / MUTEZ, orig_pymnt_amnt / MUTEZ, pymnt_amnt / MUTEZ))
+                pymnt_amnt = max(pymnt_amnt - burn_fee, 0)  # ensure not less than 0
+                logger.info("Payment to {} script requires {:.0f} gas * {:.2f} mutez-per-gas + {:10.6f} burn fee; Payment reduced from {:10.6f} to {:10.6f}".format(
+                            payment_item.paymentaddress, gas_limit, MUTEZ_PER_GAS_UNIT, burn_fee / MUTEZ, orig_pymnt_amnt / MUTEZ, pymnt_amnt / MUTEZ))
 
```

Comment:
There is one more check which is **specific to KT1 accounts**. Since fees can be really large even for small amounts, we would like to avoid payments with **total_fee > pymnt_amnt**. This was indirectly included in the previous logic, but needs to be implemented explictly now with a small if statement.

Reply:
Isn't it true that the same safety check would have been done by the following code (lines 436-438)?

```
# Subtract transaction's fee from the payment amount if delegator has to pay for it
if self.delegator_pays_xfer_fee:
    pymnt_amnt = max(pymnt_amnt - tx_fee, 0)  # ensure not less than 0
```

Reply:
The rational is that:

- KT1 address: subtract burn fee
- tz1 address: subtract reactivation fee

For both:
- subtract the transaction fee, and check if amount > 0

Reply:
This is an extra check specific for KT1 addresses. A transaction to KT1 addresses can cost much more than a tx to t1 address (due to more gas and storage consumption). We would like to avoid such transactions with 0.1 xtz fees for 0.0001 payment amount for example. This will never happen with tz1 addresses but is possible for a costly KT1 contract.

Reply:
The check for that is `if total_fee > FEE_LIMIT_CONTRACTS` right? Your edit added `if total_fee > pymnt_amnt` which is the same as done aftewards.

In my edits, KT1 did this: 
- total_fee = tx_fee + burn_fee
- check total_fee > FEE_LIMIT_CONTRACTS
- pymnt_amount = orig_amount - burn_fee

After that, for both KT1 and tz1 addresses, check that:

`pymnt_amnt = max(pymnt_amnt - tx_fee, 0)` which is the same as:

`pymnt_amnt = max(orgin_amount - burn_fee - tx_fee, 0)` or `pymnt_amnt = max(orgin_amount - total_fee, 0)`.

So I think the safety check is already done aftewards. Am I wrong?

Reply:
With your code, this check would happen only if `delegator_pays_xfer_fee` is set to `True`, right? We would like to make sure that this check happens for KT1 accounts even if `delegator_pays_xfer_fee` is set to `False` if `KT1_FEE_SAFETY_CHECK` is `True`

Reply:
Now I get it, thanks. Fine for me.